test/cxx_api_tests/LoadSBMLOptionsTests.cpp

```diff
@@ -49,61 +49,202 @@ public :
+    ASSERT_FALSE(isLess);
+    ASSERT_FALSE(isDefault);
+    ASSERT_FALSE(isAggressive);
+}
+
+TEST_F(LoadSBMLOptionsTests, LLJITOptimizationLevelLess) {
+    LoadSBMLOptions opt;
+    opt.setLLJitOptimizationLevel(LoadSBMLOptions::LLJIT_OPTIMIZATION_LEVELS::LESS);
+    bool isNone = opt.modelGeneratorOpt & LoadSBMLOptions::NONE;
+    bool isLess = opt.modelGeneratorOpt & LoadSBMLOptions::LESS;
+    bool isDefault = opt.modelGeneratorOpt & LoadSBMLOptions::DEFAULT;
+    bool isAggressive = opt.modelGeneratorOpt & LoadSBMLOptions::AGGRESSIVE;
+    ASSERT_FALSE(isNone);
+    ASSERT_TRUE(isLess);
+    ASSERT_FALSE(isDefault);
+    ASSERT_FALSE(isAggressive);
+
+}
+
+TEST_F(LoadSBMLOptionsTests, LLJITOptimizationLevelDefault) {
+    LoadSBMLOptions opt;
+    opt.setLLJitOptimizationLevel(LoadSBMLOptions::LLJIT_OPTIMIZATION_LEVELS::DEFAULT);
+    bool isNone = opt.modelGeneratorOpt & LoadSBMLOptions::NONE;
+    bool isLess = opt.modelGeneratorOpt & LoadSBMLOptions::LESS;
+    bool isDefault = opt.modelGeneratorOpt & LoadSBMLOptions::DEFAULT;
+    bool isAggressive = opt.modelGeneratorOpt & LoadSBMLOptions::AGGRESSIVE;
+    ASSERT_FALSE(isNone);
+    ASSERT_FALSE(isLess);
+    ASSERT_TRUE(isDefault);
+    ASSERT_FALSE(isAggressive);
+
+}
+
+TEST_F(LoadSBMLOptionsTests, LLJITOptimizationLevelAggressive) {
+    LoadSBMLOptions opt;
+    opt.setLLJitOptimizationLevel(LoadSBMLOptions::LLJIT_OPTIMIZATION_LEVELS::AGGRESSIVE);
+    bool isNone = opt.modelGeneratorOpt & LoadSBMLOptions::NONE;
+    bool isLess = opt.modelGeneratorOpt & LoadSBMLOptions::LESS;
+    bool isDefault = opt.modelGeneratorOpt & LoadSBMLOptions::DEFAULT;
+    bool isAggressive = opt.modelGeneratorOpt & LoadSBMLOptions::AGGRESSIVE;
+    ASSERT_FALSE(isNone);
+    ASSERT_FALSE(isLess);
+    ASSERT_FALSE(isDefault);
+    ASSERT_TRUE(isAggressive);
+
+}
+
+/**
+ * LoadSBMLOptions are updated from the global Config.
+ */
+TEST_F(LoadSBMLOptionsTests, LLJITOptimizationLevelAggressiveFromConfig) {
+    Config::setValue(Config::LLJIT_OPTIMIZATION_LEVEL, Config::LLJIT_OPTIMIZATION_LEVELS::AGGRESSIVE);
+    LoadSBMLOptions opt;
+    bool isNone = opt.modelGeneratorOpt & LoadSBMLOptions::NONE;
+    bool isLess = opt.modelGeneratorOpt & LoadSBMLOptions::LESS;
+    bool isDefault = opt.modelGeneratorOpt & LoadSBMLOptions::DEFAULT;
+    bool isAggressive = opt.modelGeneratorOpt & LoadSBMLOptions::AGGRESSIVE;
+    ASSERT_FALSE(isNone);
+    ASSERT_FALSE(isLess);
+    ASSERT_FALSE(isDefault);
+    ASSERT_TRUE(isAggressive);
+}
+
+/**
+ * modelGeneratorOptions is a bit mask system for eloquantly
+ * storing boolean options. However this is not a boolean value
+ * which is a complication. Therefore, instead of having the usual
+ * Config-which-sets-the-default, we only have the Config version.
+ */
+TEST_F(LoadSBMLOptionsTests, DISABLED_LLJitNumThreads) {
+//    LoadSBMLOptions opt;
+//    opt.setItem("LLJit_NUM_THREADS", 6);
+//    ASSERT_EQ(6, opt.getItem("LLJit_NUM_THREADS").getAs<int>());
+}
```

Comment:
This test is entirely commented out.  Do you need it?

Reply:
No I've now deleted this on my latest (uncommited) branch. The problem was `LoadSBMLOptions` cannot hold anything other than boolean, since it uses bitmasks. Therefore, we fall back on using the global `Config` options instead of allowing the user to pass in the option using `LoadSBMLOptions`. This is actually a bit of a shame because the other build options are all accessible via `LoadSBMLOptions`.